src/tla/__rewire_sequences_ext_in_apalache.tla

```diff
@@ -0,0 +1,470 @@
+------------------------ MODULE SequencesExt ---------------------------------
+\*------ MODULE __rewire_sequences_ext_in_apalache -----------------------
+(**
+ * ^^^^^^^^^^^^^^^^^^^^^^ We have to call this module SequencesExt in any
+ * case, otherwise, SANY complains.
+ *
+ * This file contains alternative definitions for the operators defined in
+ * SequencesExt. Most importantly, we are adding type annotations. We also
+ * define the Apalache-compatible behavior.
+ *
+ * These definitions are automatically rewired by the Apalache importer.
+ *
+ * Compare with the original definitions in SequencesExt.tla:
+ *
+ * https://github.com/tlaplus/CommunityModules/blob/master/modules/SequencesExt.tla
+ *)
+
+LOCAL INSTANCE Integers
+LOCAL INSTANCE Sequences
+LOCAL INSTANCE FiniteSets
+LOCAL INSTANCE __apalache_folds
+LOCAL INSTANCE __apalache_internal
+
+(**
+ * The image of the given sequence s. Cardinality(ToSet(s)) <= Len(s)
+ * see https://en.wikipedia.org/wiki/Image_(mathematics)
+ *
+ * @type: Seq(a) => Set(a);
+ *)
+ToSet(s) ==
+  { s[__i] : __i \in DOMAIN s }
+
+(**
+ * Convert a set to some sequence that contains all the elements of the
+ * set exactly once, and contains no other elements.
+ *
+ * @type: Set(a) => Seq(a);
+ *)
+SetToSeq(S) == 
+  LET __add_to_seq(seq, elem) == Append(seq, elem) IN
+  __ApalacheFoldSet(__add_to_seq, <<>>, S)
+
+(**
+ * Convert a set to a sorted sequence that contains all the elements of
+ * the set exactly once, and contains no other elements.
+ *
+ * @type: (Set(a), (a, a) => Bool) => Seq(a);
+ *)
+SetToSortSeq(S, LessThan(_, _)) ==
+  \* insert a new element element 
+  LET \* @type: (Seq(b), b) => Seq(b);
+        __insert_sorted(__seq, __newElem) ==
+    LET __next_len == Len(__seq) + 1 IN
+    \* find the position for inserting the element at
+    LET __insertion_pos ==
+      CHOOSE __p \in (DOMAIN __seq) \union { __next_len }:
+        \A __i \in DOMAIN __seq:
+          __i < __p <=> LessThan(__seq[__i], __newElem)
+    IN
+    \* copy the sequence elements or insert a new one
+    LET __copy_or_set(__i) ==
+      IF __i < __insertion_pos
+      THEN __seq[__i]
+      ELSE IF __i = __insertion_pos
+           THEN __newElem
+           ELSE __seq[__i - 1]
+    IN
+    \* raw_seq may be longer than required
+    LET __raw_seq == 
+      __ApalacheMkSeq(__ApalacheSeqCapacity(__seq) + 1, __copy_or_set)
```

Comment:
I don't understand when this would be the case? Are we allocating empty `Seq`s with positive capacity?

Reply:
Yeah, it's tricky. I did not have `SubSeq` originally and Apalache was giving me a weird counterexample. Since we are increasing capacity at every iteration, it can go beyond the length, if the set contains duplicates. I cannot say that I fully understand this example.